pytradfri/device/light.py

```diff
@@ -24,63 +33,72 @@ class Light:
     pdf
     """
 
-    def __init__(self, device, index):
+    def __init__(self, device: Device, index: int) -> None:
         """Create object of class."""
         self.device = device
         self.index = index
 
     @property
-    def supported_features(self):
+    def supported_features(self) -> int:
         """Return supported features."""
-        return supported_features(self.raw)
+        return supported_features(cast(TYPE_RAW_SIMPLE, self.raw))
 
     @property
-    def state(self):
+    def state(self) -> bool:
         """Return device state."""
-        return self.raw.get(ATTR_DEVICE_STATE) == 1
+        return cast(TYPE_RAW_SIMPLE, self.raw).get(ATTR_DEVICE_STATE) == 1
 
     @property
-    def dimmer(self):
+    def dimmer(self) -> int | None:
         """Return dimmer if present."""
         if self.supported_features & SUPPORT_BRIGHTNESS:
-            return self.raw.get(ATTR_LIGHT_DIMMER)
+            return cast(int, cast(TYPE_RAW_SIMPLE, self.raw).get(ATTR_LIGHT_DIMMER))
+        return None
 
     @property
-    def color_temp(self):
+    def color_temp(self) -> int | None:
         """Return color temperature."""
         if self.supported_features & SUPPORT_COLOR_TEMP:
             if self.raw.get(ATTR_LIGHT_MIREDS) != 0:
-                return self.raw.get(ATTR_LIGHT_MIREDS)
+                return cast(int, cast(TYPE_RAW_SIMPLE, self.raw).get(ATTR_LIGHT_MIREDS))
+        return None
 
     @property
-    def hex_color(self):
+    def hex_color(self) -> str | None:
         """Return hex color."""
         if self.supported_features & SUPPORT_HEX_COLOR:
-            return self.raw.get(ATTR_LIGHT_COLOR_HEX)
+            return cast(str, cast(TYPE_RAW_SIMPLE, self.raw).get(ATTR_LIGHT_COLOR_HEX))
+        return None
 
     @property
-    def xy_color(self):
+    def xy_color(self) -> tuple[int, int] | None:
         """Return xy color."""
         if self.supported_features & SUPPORT_XY_COLOR:
-            return (self.raw.get(ATTR_LIGHT_COLOR_X), self.raw.get(ATTR_LIGHT_COLOR_Y))
+            return cast(
+                int, cast(TYPE_RAW_SIMPLE, self.raw).get(ATTR_LIGHT_COLOR_X)
+            ), cast(int, cast(TYPE_RAW_SIMPLE, self.raw).get(ATTR_LIGHT_COLOR_Y))
+        return None
 
     @property
-    def hsb_xy_color(self):
+    def hsb_xy_color(self) -> tuple[int, int, int, int, int]:
         """Return hsb xy color."""
         return (
-            self.raw.get(ATTR_LIGHT_COLOR_HUE),
-            self.raw.get(ATTR_LIGHT_COLOR_SATURATION),
-            self.raw.get(ATTR_LIGHT_DIMMER),
-            self.raw.get(ATTR_LIGHT_COLOR_X),
-            self.raw.get(ATTR_LIGHT_COLOR_Y),
+            cast(int, cast(TYPE_RAW_SIMPLE, self.raw).get(ATTR_LIGHT_COLOR_HUE)),
```

Comment:
Why do we change the type of raw here from what's it's typed as below in its definition? That doesn't seem correct.

Reply:
This is needed, TYPE_RAW consist of 2 subtypes
- TYPE_RAW_SIMPLE, which means the dict contains simple type str, int, None
- TYPE_RAW_LIST, which means the dict contains a list.
I decided to make these 2 subtypes as their own types, in order to quite a number of casts.

But it is actually no longer needed here.